### PR TITLE
utils_disk: update package python-libguestfs for python3

### PR DESCRIPTION
**Pull request by YanhuiMa**
ID: 1694648
Signed-off-by: yama <yama@redhat.com>

### CODE REVIEW

**Comment by luckyh** [comment]
@YanhuiMa hi, please provide the test result, thanks.

**Comment by YanhuiMa** [comment]
@luckyh Because the case which uses the package will fail caused by other issues. Could I paste the install log of the package?

 740 14:19:19 INFO | Running 'yum -y install python*-libguestfs'                      
741 14:19:20 DEBUG| [stdout] Last metadata expiration check: 0:06:21 ago on Tue Apr 23 14:12:58 2019.
742 14:19:20 DEBUG| [stderr] /usr/lib/python3.6/site-packages/dnf/module/metadata_loader.py:64: Warning: g_hash_table_unref: as    sertion 'hash_table != NULL' failed
743 14:19:20 DEBUG| [stderr]   return Modulemd.objects_from_string(modules_yaml)     
744 14:19:20 DEBUG| [stderr]                                                                                                   
745 14:19:23 DEBUG| [stdout] Dependencies resolved.                                  
746 14:19:23 DEBUG| [stdout] ================================================================================
747 14:19:23 DEBUG| [stdout]  Package                                                
748 14:19:23 DEBUG| [stdout]        Arch   Version                                  Repository          Size
749 14:19:23 DEBUG| [stdout] ================================================================================
750 14:19:23 DEBUG| [stdout] Installing:                                             
751 14:19:23 DEBUG| [stdout]  python3-libguestfs                                     
752 14:19:23 DEBUG| [stdout]        x86_64 1:1.38.4-10.module+el8.0.0+3075+09be6b65 rhel-8.0-appstream 210 k
753 14:19:23 DEBUG| [stdout]                                                         
754 14:19:23 DEBUG| [stdout] Transaction Summary                                     
755 14:19:23 DEBUG| [stdout] ================================================================================
756 14:19:23 DEBUG| [stdout] Install  1 Package                                      
757 14:19:23 DEBUG| [stdout]                                                         
758 14:19:23 DEBUG| [stdout] Total download size: 210 k                              
759 14:19:23 DEBUG| [stdout] Installed size: 1.2 M                                   
760 14:19:23 DEBUG| [stdout] Downloading Packages:                                   
761 14:19:23 DEBUG| [stdout] python3-libguestfs-1.38.4-10.module+el8.0.0+307  22 MB/s | 210 kB     00:00    
762 14:19:23 DEBUG| [stdout] --------------------------------------------------------------------------------
763 14:19:23 DEBUG| [stdout] Total                                            19 MB/s | 210 kB     00:00     
764 14:19:23 DEBUG| [stdout] Running transaction check 
765 14:19:23 DEBUG| [stdout] Transaction check succeeded.                            
766 14:19:23 DEBUG| [stdout] Running transaction test                                
767 14:19:23 DEBUG| [stdout] Transaction test succeeded.                             
768 14:19:23 DEBUG| [stdout] Running transaction                                     
769 14:19:24 DEBUG| [stdout]   Preparing        :                                                        1/1 
770 14:19:24 DEBUG| [stdout]   Installing       : python3-libguestfs-1:1.38.4-10.module+el8.0.0+3075+0   1/1 
771 14:19:24 DEBUG| [stdout]   Running scriptlet: python3-libguestfs-1:1.38.4-10.module+el8.0.0+3075+0   1/1 
772 14:19:24 DEBUG| [stdout]   Verifying        : python3-libguestfs-1:1.38.4-10.module+el8.0.0+3075+0   1/1 
773 14:19:24 DEBUG| [stdout]                                                         
774 14:19:24 DEBUG| [stdout] Installed:                                              
775 14:19:24 DEBUG| [stdout]   python3-libguestfs.x86_64 1:1.38.4-10.module+el8.0.0+3075+09be6b65            
776 14:19:24 DEBUG| [stdout]                                                                                                   
777 14:19:24 DEBUG| [stdout] Complete!                                               
778 14:19:24 INFO | Command 'yum -y install python*-libguestfs' finished with 0 after 4.52269434928894s

File: virttest/utils_disk.py [hunk 1]
```diff
@@ -1080,7 +1080,7 @@ def __init__(self, disk, backend='direct'):
         try:
             import guestfs
         except ImportError:
-            install_cmd = "yum -y install python-libguestfs"
+            install_cmd = "yum -y install python*-libguestfs"
```
Comment (PaulYuuu) [inline virttest/utils_disk.py]: Suggest to install it using `utils_package.package_install(pkg)`

File: virttest/utils_disk.py [hunk 2]
```diff
@@ -1080,9 +1081,8 @@ def __init__(self, disk, backend='direct'):
         try:
             import guestfs
         except ImportError:
-            install_cmd = "yum -y install python-libguestfs"
             try:
-                process.run(install_cmd)
+                utils_package.package_install("python*-libguestfs")
```
Comment (balamuruhans) [inline virttest/utils_disk.py]:
```python
if not utils_package.package_install("python*-libguestfs"):
    raise exceptions.TestSkipError()
try:
    import guestfs
except ImportError:
    raise exceptions.TestError()
```

Comment (YanhuiMa) [reply virttest/utils_disk.py]: @balamuruhans Updated, thanks.

**Comment by vivianQizhu** [comment]
@YanhuiMa Let's just simply abondon `package_install`, it's nothing to do with your original fix anyway.

**Comment by balamuruhans** [comment]
@vivianQizhu can we importing it locally so that it will fix it,
```python
        :: :: ::
        except ImportError:
            from virttest.utils_package import package_install
            if not package_install("python*-libguestfs"):
                raise exceptions.TestSkipError('We need python-libguestfs (or '
                                               'the equivalent for your '
                                               'distro) for this particular '
                                               'feature (modifying guest '
                                               'files with libguestfs)')
            try:
                import guestfs
            except ImportError:
raise exceptions.TestError("Couldn't import guestfs")
```

**Comment by YanhuiMa** [comment]
> @vivianQizhu can we importing it locally so that it will fix it,
> 
> ```python
>         :: :: ::
>         except ImportError:
>             from virttest.utils_package import package_install
>             if not package_install("python*-libguestfs"):
>                 raise exceptions.TestSkipError('We need python-libguestfs (or '
>                                                'the equivalent for your '
>                                                'distro) for this particular '
>                                                'feature (modifying guest '
>                                                'files with libguestfs)')
>             try:
>                 import guestfs
>             except ImportError:
> raise exceptions.TestError("Couldn't import guestfs")
> ```

The check has passed, thanks!

**Comment by balamuruhans** [comment]
Thanks @YanhuiMa @vivianQizhu, let's get it in.

**Pull request merged**